### DISCUSSION TITLE
Welcome to OpnForm Discussions!

### DISCUSSION THREAD

**Discussion by JhumanJ** [OpnForm/OpnForm#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Suggest new features.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rss2nse** [comment]
Dear JhumanJ, i saw your opnform free code web form.. Excellent work.. I would like it to use for one admission purpose.. Shall i use it? Is it possible to add or generate Unique ID for every successful submission? How come this service come free to use though others are offering charges for accepting image fileds? I really wounder. Thank you and I congratulate once again for your wounderful work!.. 

Excellent Work!

All the best

**Reply by JhumanJ** [reply]
Hey there, thanks for your kind words. We already have an option for generating unique IDs. You can do that here from the advanced options of a text field (you can choose between uuids and incremental ids). You're right hosting & file uploads are costing us money, but since the project is new it's still very reasonable. If the project grows we might offer a paid plan with more features later on to cover our costs. The project will always remain open-source and free to self-host!

![image](https://user-images.githubusercontent.com/11312432/209428604-60fdb385-e1f7-4283-a629-bbe4d2efd716.png)

**Comment by amitavroy** [comment]
Hi @JhumanJ I was trying to install Opnform on my machine and while doing that, I saw that there are some errors coming while I was trying to run the migrations.

Now, I understand that the default .env.example has DB set to postgres. But, do you support MySQL as well or there are plans to do that? Because I was planning to document the process of how to set this app up on a local instance / EC2 instance as a self-hosted option. And, while doing that I encountered this MySQL default value error. So, decided to check with you before raising as an issue.

Thanks
Amitav Roy

**Comment by JhumanJ** [comment]
Hey @amitavroy, thanks for your interest and desire to contribute. Could you please create a dedicated issue for this? Happy to work with you on making this compatible with MySQL.

Thanks!
